stats/influxdb/collector.go

```diff
@@ -132,8 +133,39 @@ func (c *Collector) commit() {
 }
 
 func (c *Collector) extractTagsToValues(tags map[string]string, values map[string]interface{}) map[string]interface{} {
+tags:
 	for _, tag := range c.Config.TagsAsFields {
 		if val, ok := tags[tag]; ok {
+			for _, t := range c.Config.IntFields {
+				if tag == t {
+					i, err := strconv.ParseInt(val, 10, 64)
```

Comment:
I do think that this linear search will be better if it uses the [sort](https://golang.org/src/sort/sort.go) package and specifically [sort.Strings](https://golang.org/src/sort/sort.go?s=8285:8309#L304) and [sort.SearchStrings](https://golang.org/pkg/sort/#SearchStrings), you can look at the [csv collector where it used](https://github.com/loadimpact/k6/blob/7e4309c3e77f2d196628d6f440da14a44a9a796c/stats/csv/collector.go)

This also will make this code half in size and you could drop the label ;)

Reply:
This probably can be done even better if you have only one sorted slice of some struct that has the name of the field and what type it is. 
```golang
struct {
tagName string
kind fieldKind
}
```
`fieldKind` being an "enum" of either int, bool or float ... (`type` is reserved word so I can't use it at least for the field name :( )

This way there will be a single search and during the construction of this struct you can report(with a warning ?) that a field is in both the int and bool fields (for example).

~~Also because this will be faster maybe we will be able to reuse it for normal tags?~~ I went ahead and checked ... tag values are always strings both in the current [1.8](https://docs.influxdata.com/influxdb/v1.8/concepts/glossary/#tag-value) and the [2.0 beta](https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/influxdb-v1/tagvalues/#tag) so there is no way to it for them :D